cpp/include/raft/sparse/distance/l2_distance.cuh

```diff
@@ -116,6 +145,55 @@ void compute_l2(value_t *out, const value_idx *Q_coo_rows,
                     expansion_func);
 }
 
+template <typename value_idx, typename value_t, int tpb = 256>
+void compute_correlation(value_t *C, const value_t *Q_sq_norms,
+                         const value_t *R_sq_norms, const value_t *Q_norms,
+                         const value_t *R_norms, value_idx n_rows,
+                         value_idx n_cols, value_idx n, cudaStream_t stream) {
+  int blocks = raft::ceildiv<size_t>((size_t)n_rows * n_cols, tpb);
+  compute_correlation_warp_kernel<<<blocks, tpb, 0, stream>>>(
+    C, Q_sq_norms, R_sq_norms, Q_norms, R_norms, n_rows, n_cols, n);
+}
+
+template <typename value_idx, typename value_t, int tpb = 256>
+void compute_corr(value_t *out, const value_idx *Q_coo_rows,
+                  const value_t *Q_data, value_idx Q_nnz,
+                  const value_idx *R_coo_rows, const value_t *R_data,
+                  value_idx R_nnz, value_idx m, value_idx n, value_idx n_cols,
+                  std::shared_ptr<raft::mr::device::allocator> alloc,
+                  cudaStream_t stream) {
+  // sum_sq for std dev
+  raft::mr::device::buffer<value_t> Q_sq_norms(alloc, stream, m);
+  raft::mr::device::buffer<value_t> R_sq_norms(alloc, stream, n);
+
+  // sum for mean
+  raft::mr::device::buffer<value_t> Q_norms(alloc, stream, m);
```

Comment:
If it wouldn't be too many changes, could you use rmm uvectors in general instead? (just adding a single comment to avoid repeated ones)

Reply:
Yeah I don't think that would be too much to do for this PR.